tests/test_context.py

```diff
@@ -0,0 +1,41 @@
+# Source pytest module
+# Copyright (C) 2020  Ngô Ngọc Đức Huy
+#
+# This file is part of palace.
+#
+# palace is free software: you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published
+# by the Free Software Foundation, either version 3 of the License,
+# or (at your option) any later version.
+#
+# palace is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with palace.  If not, see <https://www.gnu.org/licenses/>.
+
+"""This pytest module tries to test the correctness of the class Context."""
+
+from palace import Context, current_context
+
+
+def test_with_context(device):
+    """Test if `with` can be used to start a context
+    and is destroyed properly.
+    """
+    with Context(device) as context:
+        assert current_context() == context
+
+
+def test_nested_context_manager(device):
+    """Test if the context manager returns to the
+    previous context.
+    """
+    with Context(device) as ctx_1:
+        assert current_context() == ctx_1
+        with Context(device) as ctx_2:
+            assert current_context() == ctx_2
+            assert ctx_1 != ctx_2
```

Comment:
While this should pass,
1. It has nothing to do this this test
2. It relies on alure's implementation details

Don't get me wrong, I'd be happy to have this warrantied, so please just move it to another test.

Reply:
gentle ping

Reply:
Oops I only mean the `assert ctx_1 != ctx_2` line.

Reply:
But I mean the other two assertions should be checked already by the previous test, no?

Reply:
You're right.